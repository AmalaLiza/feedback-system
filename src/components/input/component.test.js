import React from "react";
import { shallow } from "enzyme";
import Input from "./component";
import renderer from "react-test-renderer";

const props = {
	label: "label",
	handleChange: jest.fn(),
	type: "text",
	placeHolder: "Type here",
	classes: {},
};

function setup() {
	const wrapper = shallow(<Input {...props} />);
	return { wrapper, props };
}

describe("Input Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<Input {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should have an input", () => {
		const { wrapper } = setup();
		expect(wrapper.find("input").exists()).toBe(true);
	});

	it("should call handleChange", () => {
		const handleChange = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: "the-value" },
		};
		const component = shallow(
			<Input label="label" handleChange={handleChange} />
		);
		component.find("input").simulate("change", event);
		expect(handleChange).toBeCalledWith("the-value");
	});

	it("should call handleChange", () => {
		const handleChange = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: "the-value" },
		};
		const component = shallow(
			<Input label="label" handleChange={handleChange} />
		);
		component.find("input").simulate("focus", event);
		expect(component.state().active).toBe(true);
		component.find("input").simulate("focus", event);
		expect(component.state().active).toBe(false);
	});
});
