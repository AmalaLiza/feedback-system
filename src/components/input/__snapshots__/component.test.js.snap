// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Test Suite renders correctly 1`] = `
<div
  className="container"
>
  <label
    className="label"
    htmlFor="label"
  >
    label
  </label>
  <input
    className="input"
    id="label"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Type here"
    type="text"
    value=""
  />
</div>
`;
